### PR TITLE
scatter: fix the unexpected error code

### PR DESCRIPTION
**Pull request by nolouch**
### What problem does this PR solve?

Issue Number: Close https://github.com/tikv/pd/issues/7073

### What is changed and how does it work?

```commit-message
scatter: fix the unexpected error code
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

File: server/grpc_service.go [hunk 1]
```diff
@@ -1710,13 +1710,6 @@ func (s *GrpcServer) ScatterRegion(ctx context.Context, request *pdpb.ScatterReg
 		return nil, err
 	}
 
-	if op == nil {
```
Comment (nolouch) [inline server/grpc_service.go:1713]: Should be seen as a normal situation

File: pkg/schedule/scatter/region_scatterer.go [hunk 2]
```diff
@@ -296,7 +296,7 @@ func (r *RegionScatterer) scatterRegion(region *core.RegionInfo, group string, s
 	for _, peer := range region.GetPeers() {
 		store := r.cluster.GetStore(peer.GetStoreId())
 		if store == nil {
-			return nil
+			return nil, errs.ErrGetSourceStore.FastGenByArgs(fmt.Sprintf("store not found, peer: %v", peer))
```
Comment (bufferflies) [inline pkg/schedule/scatter/region_scatterer.go:299]: how about add region id for debug? In this case, should we need to tell users wait to fix the fault peer .

File: pkg/schedule/scatter/region_scatterer.go [hunk 3]
```diff
@@ -384,8 +384,7 @@ func (r *RegionScatterer) scatterRegion(region *core.RegionInfo, group string, s
 			targetPeers[peer.GetStoreId()] = peer
 		}
 		r.Put(targetPeers, region.GetLeader().GetStoreId(), group)
-		log.Debug("fail to create scatter region operator", errs.ZapError(err))
-		return nil
+		return nil, errs.ErrCreateOperator.FastGenByArgs(fmt.Sprintf("failed to create scatter region operator for region %v", region.GetID()))
```
Comment (CabinfeverB) [inline pkg/schedule/scatter/region_scatterer.go:387]: I prefer to print the error for debug

File: pkg/schedule/scatter/region_scatterer_test.go [hunk 4]
```diff
@@ -140,6 +146,7 @@ func scatter(re *require.Assertions, numStores, numRegions uint64, useRules bool
 		re.LessOrEqual(float64(count), 1.1*float64(numRegions)/float64(numStores))
 		re.GreaterOrEqual(float64(count), 0.9*float64(numRegions)/float64(numStores))
 	}
+	re.GreaterOrEqual(noNeedMoveNum, 0)
```
Comment (CabinfeverB) [inline pkg/schedule/scatter/region_scatterer_test.go:149]: It seems always be true

**Review by CabinfeverB — COMMENTED** [review]
REST LGTM

File: server/grpc_service.go [hunk 5]
```diff
@@ -1709,18 +1709,13 @@ func (s *GrpcServer) ScatterRegion(ctx context.Context, request *pdpb.ScatterReg
...
-		}, nil
-	}
-
-	if !rc.GetOperatorController().AddOperator(op) {
-		return &pdpb.ScatterRegionResponse{
-			Header: s.wrapErrorToHeader(pdpb.ErrorType_UNKNOWN,
-				"operator cancelled because store limit exceeded"),
-		}, nil
+	if op != nil {
+		if !rc.GetOperatorController().AddOperator(op) {
```
Comment (rleungx) [inline server/grpc_service.go:1713]: I realize that if the operator cannot be added, the store limit exceeded is just one of the reasons.

File: pkg/schedule/scatter/region_scatterer_test.go (hunk 4 above)
Comment (nolouch) [reply pkg/schedule/scatter/region_scatterer_test.go:149]: yes, here only verify `noNeedMoveNum` is normal situation and no error reported.

File: server/grpc_service.go (hunk 5 above)
Comment (nolouch) [reply server/grpc_service.go:1713]: What other reasons?

Comment (rleungx) [reply server/grpc_service.go:1713]: If checkAddOperator fails, it will also return false.

Comment (nolouch) [reply server/grpc_service.go:1713]: I changed the error info. PTAL @rleungx

**Review by rleungx — APPROVED** [review]

**Review by CabinfeverB — APPROVED** [review]

**Pull request merged**